catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -150,20 +140,141 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
+            String message = "Error getting Metacard.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
 
         try {
-            Handlebars handlebars = new Handlebars(templateLoader);
-            Template template = handlebars.compile("report");
-            html = template.apply(registryMap);
+            html = registryReportBuilder.getSummaryHtmlFromMetacard(metacard);
         } catch (IOException e) {
-            LOGGER.error("Error compiling and applying report template.");
+            String message = "Error in compiling and applying report template.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
         }
 
         return Response.ok(html)
                 .build();
     }
 
+    @Path("/{registryId}/report/{section}")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewSectionInfoHtml(@PathParam("registryId") final String registryId,
+            @QueryParam("sourceId") final List<String> sourceIds,
+            @PathParam("section") final String section) {
+        String html = "";
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            Set<String> sourceIdsSet = new HashSet<>();
+            if (CollectionUtils.isNotEmpty(sourceIds)) {
+                sourceIdsSet.addAll(sourceIds);
+            }
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIdsSet);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For registry id: '{}', optional sources: {}",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For registry id: '{}', optional source ids: {}.",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(message)
+                    .build();
+        }
+
+        List<String> validSections = Arrays.asList(registryReportBuilder.ORGANIZATIONS);
+        if (!isValidSection(validSections, section)) {
```

Comment:
We should probably move this validation up next to the other request validation (the one checking registryId) so bad requests can be tested before any calls to federation admin service.